daml-lf/archive/src/test/scala/com/digitalasset/daml/lf/archive/DecodeV1Spec.scala

```diff
@@ -1535,6 +1570,46 @@ class DecodeV1Spec
         )
       }
     }
+
+    s"decode upgradedPackageId iff version >= ${LV.Features.packageUpgrades} " in {
+      forEveryVersionSuchThat(_ >= LV.Features.packageMetadata) { version =>
```

Comment:
Is there a reason that we don't check for a parse failure in versions without the packageMetadata here, like we did with the `decode softFetch iff version` test?

Reply:
yes exactly, prior to `LV.Features.packageMetadata` any metadata would be a parse error, so we don't care about a subfield